### PR TITLE
Adds label to DateSelect

### LINKED ISSUE #1991: Bug: DateSelect ignores label prop
Hedgey form "Unlock Date" label is ignored.

Image

Needs to avoid always showing label, as this is included in the DateRange input that only contains one label.

### PR DESCRIPTION
**Pull request by plor**
## GitHub Issue

fixes #1991

## Changes

* also removes "_var" from unused warnings
 * DateRange excludes passing label into DateSelect

Screenshot from 2022-07-20 13-51-26

## Packages Added

No packages, but I modified eslint config

## Checks

Before making your PR, please check the following:

- [x] Critical lint errors are resolved
- [x] App runs and builds locally

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -45,7 +45,7 @@ module.exports = {
   rules: {
     // default rules
     'no-console': 'off',
-    'no-unused-vars': 'warn', // change to 'error' 🙏
+    'no-unused-vars': ['warn', { "varsIgnorePattern": "^_" }], // change to 'error' 🙏
```
Comment (jordanlesich) [inline .eslintrc.js:48]: Curious about this

File: src/formBuilder/dateRange.jsx [hunk 2]
```diff
@@ -10,11 +10,14 @@ const DateRange = props => {
   const today = new Date();
   const startDate = watch('startDate');
 
+  // remove label prop from date selects
+  const { label: _l, ...selectProps } = props;
```
Comment (jordanlesich) [inline src/formBuilder/dateRange.jsx:14]: Very nice

**Review by jordanlesich — APPROVED** [review]
The solution looks great to me. Don't have time to QA it, but if it works, I'd this is ready to rock. 

I'm going to guess that eslint rule may have been to allow unused vars so long as they have an underscore?

File: .eslintrc.js (hunk 1 above)
Comment (plor) [reply .eslintrc.js:48]: Yeah, allows you to do "_ignoreMe" to split up objects as I did to pull out the label. I know in `go` you just use a `_` variable for this.

**Pull request merged**